From a957827bee58b456595907e93f96b40145fbcb33 Mon Sep 17 00:00:00 2001
From: Emmanuel Jillela <emmanuel.jillela@intel.com>
Date: Thu, 19 Nov 2015 13:09:43 -0700
Subject: [PATCH]    Title: ifconfig  up and than down and up again sequence
 does kernel crashe in ATOM datapipe    Issue: <HSD:244486> 
   IssueType: <BUG>    Description: <clearing ARM11 bridge
 interruptin netip susbsystem ISR.>    Reviewers: <chandrap,
 ejillela>

---
 arch/x86/NetIP_SubSystem/netip_subsystem_defs.h |    2 +-
 arch/x86/NetIP_SubSystem/netip_subsystem_lld.c  |    8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/arch/x86/NetIP_SubSystem/netip_subsystem_defs.h b/arch/x86/NetIP_SubSystem/netip_subsystem_defs.h
index bb072d3..4708577 100644
--- a/arch/x86/NetIP_SubSystem/netip_subsystem_defs.h
+++ b/arch/x86/NetIP_SubSystem/netip_subsystem_defs.h
@@ -43,7 +43,7 @@
 #define NETSS_DBG_PRINT(fmt, args...) ; 
 #endif
 
-#define NETSS_MAX_ARM11_INTR 64
+#define NETSS_MAX_ARM11_INTR 96
 
 
 struct net_subsystem_drv_data {
diff --git a/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c b/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c
index 72483e4..54d65e1 100755
--- a/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c
+++ b/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c
@@ -289,12 +289,16 @@ static irqreturn_t net_subsystem_isr(int irq, void *dev_id)
                         {
                             pnet_ss->arm11_irqs[j].func(irq, pnet_ss->arm11_irqs[j].args);
                             /*Arm11 bottom half will clear the NetIP bridge interrupt */
-                            //ack_intr |= (1 << i);
+                            ack_intr |= (1 << i);
+                            NETSS_DBG_PRINT("Processed Arm 11 interrupt %d\n", j);
                         }
                         else
                         {
                             pnet_ss->pending_arm11_interrupts |=  (1 << j);
-                            printk("No irq registered for ARM11 interrupt %d, Marking it pending\n", j);
+                            avalanche_intc_disable_irq(j);
+                            ack_intr |= (1 << i);
+                            NETSS_DBG_PRINT("No irq registered for ARM11 interrupt %d, Marking it pending\n", j) ;
+
                         }
                         /* no need to clear ATOM_INTC here avalanche_intc_clear_status(j); */
                         /* status clear need to be done by requester */
-- 
1.7.9.5

